### ISSUE TITLE
Raspberry 4: OMV 2019-07-17 does not show WEB UI (nginx -t fails)

### ISSUE THREAD

**Issue by PieBru** [openmediavault/openmediavault#425]
Hi,
suspecting a fault on my side I originally [posted](https://forum.openmediavault.org/index.php/Thread/28476-SOLVED-Raspberry-4-OMV-2019-07-17-does-not-show-WEB-UI-nginx-t-fails/) on the forum. After investigating a bit I found a solution, but IMO this problem should be fixed at the origin by the dev team.

**How to reproduce**: OMV4 [v2019-07-17](https://sourceforge.net/projects/openmediavault/files/Raspberry%20Pi%20images/OMV_4_Raspberry_Pi_2_3_3Plus_4.img.xz/download) on a RasPI4 boots but nginx fails to start at least on an IPv4 network, thus leaving the user without the UI and sometime erroneously thinking OMV didn't boot.

**The problem**: nginx try to open port 80 both on IPv4 and IPv6, but the system returns error.

**A temporary solution**: as I don't have IPv6 out of my LAN, my lazy solution was to comment-out the IPv6 port definition and enable `listen 80;` on both `/etc/nginx/sites-available/openmediavault-webgui` and `/etc/nginx/sites-available/default`.
This way `systemctl start nginx` worked and thw WEB UI appeared.

**A more correct approach** is explained [here](https://stackoverflow.com/questions/14972792/nginx-nginx-emerg-bind-to-80-failed-98-address-already-in-use#15101745).

Thanks,
Piero

**Comment by votdev** [comment]
The nginx mkconf script already has a check if IPv6 is enabled. Can you please check
```
$ ls -alh /proc/net/if_inet6
$ cat /proc/net/if_inet6
$ grep --count --invert-match 'lo$' /proc/net/if_inet6
```

**Comment by PieBru** [comment]
Here it is, freshly etched plus my two commented IPv6 lines and this morning `apt update && apt full-upgrade` .
Please note also the `omv-initsystem` reply.

```
root@openmediavault.local's password: 
 ____                 _                          ____  _   _  _   
|  _ \ __ _ ___ _ __ | |__   ___ _ __ _ __ _   _|  _ \(_) | || |  
| |_) / _` / __| '_ \| '_ \ / _ \ '__| '__| | | | |_) | | | || |_ 
|  _ < (_| \__ \ |_) | |_) |  __/ |  | |  | |_| |  __/| | |__   _|
|_| \_\__,_|___/ .__/|_.__/ \___|_|  |_|   \__, |_|   |_|    |_|  
               |_|                         |___/                  

Welcome to ARMBIAN 5.46 experimental Debian GNU/Linux 9 (stretch) 4.19.50-v7l+   
System load:   0.04 0.04 0.01  	Up time:       2 min		
Memory usage:  2 % of 3906MB 	IP:            172.16.1.6
CPU temp:      48°C           	
Usage of /:    17% of 7.2G   	

You have new mail.
Last login: Fri Aug 30 10:51:09 2019 from 172.16.2.127

root@openmediavault:~# omv-initsystem 
/boot/initrd.img-4.14.47-rockchip does not exist. Cannot update.
root@openmediavault:~# ls -alh /proc/net/if_inet6
ls: cannot access '/proc/net/if_inet6': No such file or directory
root@openmediavault:~# grep --count --invert-match 'lo$' /proc/net/if_inet6
grep: /proc/net/if_inet6: No such file or directory
root@openmediavault:~# 
```

**Comment by PieBru** [comment]
After restoring the original two above mentioned config files the result is the same.
Thanks,
Piero

**Comment by votdev** [comment]
Can not reproduce this with a core OMV4 system (don't know what the ARMBIAN image is doing beside the OMV functionality).

> root@openmediavault:~# ls -alh /proc/net/if_inet6
> ls: cannot access '/proc/net/if_inet6': No such file or directory

Based on your output the file does not exist, so the ``omv_is_ipv6_enabled`` (see [here](https://github.com/openmediavault/openmediavault/blob/4.x/deb/openmediavault/usr/share/openmediavault/scripts/helper-functions#L825)) fails ([here](https://github.com/openmediavault/openmediavault/blob/4.x/deb/openmediavault/usr/share/openmediavault/mkconf/nginx.d/10webgui#L58)) and writes the ``listen 80;``line ([here](https://github.com/openmediavault/openmediavault/blob/4.x/deb/openmediavault/usr/share/openmediavault/mkconf/nginx.d/10webgui#L105)).

**Comment by PieBru** [comment]
Just my 2 cents...

A drastic solution exists: **stop distributing Armbian-packaged OMV ready-to-use not working SD images**. Then suggest to the users-base the manual OMV installation over a restricted list of OSs, thus leaving to the users all the OS issues, however leveraging the users skill level required.

Sorry, but I don't understand why you closed this **unresolved** issue, it's about an official OMV-distributed SD image, where the OMV team - not a random user - choosed Armbian.

BTW, reproducing it should be easy and economical feasible without hurrying up a reply: anyone having a Raspberry and a SD card may reproduce this issue from the first boot.

Also, I  noted IPv6 is disabled by default in the OMV WEB UI, so disabling it also in the .conf should be enough to avoid users escape to other solutions at the first Raspberry boot.

Piero

**Comment by ryecoaaron** [comment]
@PieBru Don't assume this is a problem with all arm images.  The RPi images are not a real armbian image.  This image is not good on RPi4s but I didn't want to remove it until I had a better one since it fine on older RPis.  And this isn't an official OMV image.  Volker just lets me use the same host.  I am trying to build a pure Raspbian image but just haven't had much time.

If you want OMV on your RPi4, use the Raspbian Buster Lite image.  Yes, this forces you to use OMV 5.x in its beta phase but I don't have a better solution.  We are a small team and the rpi4 has only been out a little over a month and I just got one a couple of weeks ago.

**Comment by PieBru** [comment]
Thanks for your clarification @ryecoaaron , I will give a try to Raspi4 + RaspbianBuster + OMV5beta.

Sorry @votdev , I reacted to what I perceived was showed through the OMV pages.

OMV is a great and promising project, and IMO the Raspi4 with 4GB RAM could entail an excellent evolution because with 4GB RAM it's far most effective than its predecessors, maybe except for heavy encoding/transcoding.

Thanks all,
Piero
